datafusion/physical-plan/src/projection.rs

```diff
@@ -308,8 +470,11 @@ impl ExecutionPlan for ProjectionExec {
         self: Arc<Self>,
         mut children: Vec<Arc<dyn ExecutionPlan>>,
     ) -> Result<Arc<dyn ExecutionPlan>> {
-        ProjectionExec::try_new(self.expr.clone(), children.swap_remove(0))
-            .map(|p| Arc::new(p) as _)
+        ProjectionExec::try_new(
+            self.projection.as_ref().to_vec(),
```

Comment:
If you made Projection implement into_iterator over ProjectionExpr, you could probably simply pass in `self.projection.clone()` here which I think would make for a nice interface